[Yeo Jun Jie] iP

yeojunjie opened the pull request:
# Duke

## Dukescription

Duke is able to do some *things* to help you manage your schoolwork. Some of the *things* include:
* adding tasks
* deleting tasks ~to escape from them~
* completing tasks
* searching for tasks

There are many types of ~pain~ tasks. Duke supports
1. Events
2. Deadlines
3. To-dos

## Features
- [X] Manage tasks
- [ ] Order tasks by deadline (coming soon)
- [ ] A help menu which appears when the user types `help` or when the user input is, let's just say, open to interpretation. (coming soon)

## Cancelled Features

- [ ] ~A personality. In contrast to [Sir Duke](https://www.youtube.com/watch?v=6sIjSNTS7Fs), this program will take on the personality of an Encik. We call it an Encik Duke.~
This idea was rejected because it uses a very specialised vocabulary, which would not be appropriate for international audiences. 🫤 An example would be the following phrase.
    > one times good one

```
// This is a comment.
/* This is also a comment. */
```

yeojunjie committed:
Add level 1 features

yeojunjie committed:
Add a comment

yeojunjie committed:
Add two comments

yeojunjie committed:
Remove unnecessary comments

yeojunjie committed:
Added level 2 features

yeojunjie committed:
Add Task class and level 3 features

yeojunjie committed:
Add features for level 4

yeojunjie committed:
Added automated text UI testing

yeojunjie committed:
Handle errors caused by bad input for `mark` and `unmark`

yeojunjie committed:
Add features for level 6

yeojunjie committed:
Add ability to read from disk

yeojunjie committed:
Add function to save to disk

yeojunjie committed:
Use java.time.LocalDateTime

yeojunjie committed:
Merge branch 'branch-Level-8'

yeojunjie committed:
Refactored the code into classes

yeojunjie committed:
Organize the classes into java packages

yeojunjie committed:
Add configuration files for Gradle

yeojunjie committed:
Configure junit and add two unit tests

yeojunjie committed:
Fix bug that creates duke.txt/ instead of duke.txt

yeojunjie committed:
Add javadocs

yeojunjie committed:
Enforce coding standard

yeojunjie committed:
Add search feature

yeojunjie committed:
Merge branch 'branch-A-CodingStandard' into branch-Level-9

yeojunjie committed:
Revert "Merge branch 'branch-A-CodingStandard' into branch-Level-9"

This reverts commit e2261a332374dc37e4171a088c05514ddd5c8a4d.

yeojunjie committed:
Merge branch 'branch-A-CodingStandard'

yeojunjie committed:
Merge branch 'branch-Level-9'

yeojunjie committed:
Adhere to code style

boredcoco commented on data/duke.txt:
```diff
@@ -0,0 +1,9 @@
+T|1|read book
+D|1|return book|2022-02-02T22:22
```
Perhaps this file should be deleted before committing so that the user can test the code to see if it works without a data/duke.txt file being generated at first?

boredcoco commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,29 @@
+package duke;
+
```
Consider creating a separate package for all classes that extend Task within the Duke package.

boredcoco commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,26 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ParserTest {
+    @Test
+    public void getCommandType_incorrectInput_nonsenseRecognized() {
+        TaskList emptyTaskList = new TaskList();
+        Parser parser = new Parser(emptyTaskList);
+        assertEquals(parser.getCommandType("bruospdk"), CommandType.NONSENSE);
+    }
+
+    @Test
+    public void execute_validEventAddition_eventIsAdded() {
+        TaskList taskList = new TaskList();
+        Parser parser = new Parser(taskList);
+        parser.execute("event New Year's Day 2023 @ 2023-01-01T00:00:00");
+        assertEquals(taskList.size(), 1);
+    }
+}
```
Perhaps you can add more test cases?

boredcoco commented on src/main/java/duke/CommandType.java:
```diff
@@ -0,0 +1,12 @@
+package duke;
+
+public enum CommandType {
+    EXIT,
+    ADD,
+    REMOVE,
+    MARK_AS_DONE,
+    MARK_AS_UNDONE,
+    SEARCH,
+    LIST,
+    NONSENSE
+}
```
I like the use of enums here.

boredcoco commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,52 @@
+ */
+public class Duke {
+
+    private Storage storage;
+    private TaskList taskList;
+    private Ui ui;
```
I like how you extracted out the UI and various components, good job!

boredcoco commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,29 @@
+package duke;
```
Like Deadline, you can consider putting this in a separate package within the Duke package so that it is more organised.

boredcoco commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,128 @@
+     * @param userCommand the command entered by the user.
+     * @return            the type of the command, as a CommandType
+     */
+    public CommandType getCommandType(String userCommand) {
+        if (userCommand.startsWith("bye")) {
+            return CommandType.EXIT;
+        } else if (userCommand.startsWith("todo") ||
+                   userCommand.startsWith("event") ||
+                   userCommand.startsWith("deadline")) {
+            return CommandType.ADD;
+        } else if (userCommand.startsWith("remove")) {
+            return CommandType.REMOVE;
+        } else if (userCommand.startsWith("mark")) {
+            return CommandType.MARK_AS_DONE;
+        } else if (userCommand.startsWith("unmark")) {
+            return CommandType.MARK_AS_UNDONE;
+        } else if (userCommand.startsWith("list")) {
+            return CommandType.LIST;
+        } else if (userCommand.startsWith("search")) {
+            return CommandType.SEARCH;
+        } else {
+            return CommandType.NONSENSE;
+        }
+    }
```
I like the use of enums here, good job!

boredcoco commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,107 @@
+package duke;
+
+import duke.TaskList;
+import duke.ToDo;
+import duke.Deadline;
+import duke.Event;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Objects;
+import java.util.Scanner;
```
I like the way you organised your imports so it is neater and easier to read, good job!

boredcoco reviewed (commented):
Overall, I like the way you tried to incorporate SWE principles into IP. Although at this point of the review you have yet to start on Level-10, thus far your project looks great! You could incorporate more test cases and put all classes that inherit Task into a separate package for better organisation. I hope the suggestions given are useful.

yeojunjie committed:
Use JavaFX to implement a GUI

yeojunjie committed:
Use JavaFX to create a GUI

yeojunjie committed:
Revert "Use JavaFX to create a GUI"

This reverts commit 371f1249a28362d890809ace5f65c6a393aa1726.

yeojunjie committed:
Revert "Revert "Use JavaFX to create a GUI""

This reverts commit 0e43f494cf4f73ddcac9f494a9d0525cc0080132.

yeojunjie committed:
Revert "Revert "Revert "Use JavaFX to create a GUI"""

This reverts commit 7a3de8cc09df10a6591c0947acf91875336fd394.

yeojunjie committed:
Merge branch 'master' of https://github.com/yeojunjie/ip

yeojunjie committed:
Include shadow as a plugin for Gradle so that fat jar files can be generated

yeojunjie committed:
Add assertions

yeojunjie committed:
Add assertion

There are three types of tasks that the user can add.

We assert the assumption that a task that is neither a deadline nor an event is a todo.

yeojunjie committed:
Use String.format() to clarify encoding scheme

yeojunjie committed:
Fix bug: Task list not saved upon exit

yeojunjie committed:
Merge branch 'branch-A-CodeQuality'

yeojunjie committed:
Revert "Merge branch 'branch-A-CodeQuality'"

This reverts commit 41a1bb9a9992efe0d4fa10e785d64f57859904fa, reversing
changes made to 07fc6b2ae9ebb1d3fc53900a45350ad186fbbbde.

yeojunjie committed:
Revert "Revert "Merge branch 'branch-A-CodeQuality'""

This reverts commit 5f637d42ada6c0a9a977f11dd6f53d9d252b1eef.

yeojunjie committed:
Use Streams

yeojunjie committed:
Merge pull request #1 from yeojunjie/branch-A-Streams

Use Streams

yeojunjie committed:
Add ability to sort tasks by name

yeojunjie committed:
Abstract away the listing of tasks to TaskList::toDisplayString

yeojunjie committed:
Merge pull request #2 from yeojunjie/branch-C-Sort

Add ability to sort tasks by name

yeojunjie committed:
Update README documentation

yeojunjie committed:
Add Ui.png

yeojunjie committed:
Replace images

The new photos require no attribution.

yeojunjie committed:
Attribute reused code

yeojunjie committed:
Reduce line length
